src/api-service/__app__/onefuzzlib/notifications/main.py

```diff
@@ -134,9 +140,15 @@ def new_files(container: Container, filename: str) -> None:
             )
             send_message(task.task_id, bytes(url, "utf-8"), StorageType.corpus)
 
-    if report:
+    if isinstance(report, Report):
```

Comment:
It doesn't look like we check if `report is None` earlier (at least there are no notifications). That means `report` could be `None`, but we send an `EventFileAdded`. I guess that's what we want, because it just means a non-report file was added to the container. But maybe it would be better to pull the notification loop out into a function, and do a "match" much earlier in this function. So, the outer structure of this function would then just dispatch based on whether `filename` is a report, regression report, or neither.

Reply:
The teams integration can be used to notify you on file changes.  It's not included in this diff, but  look at the larger file as `if not report: continue` in the `for notification in notifications` loop.

Only ADO & Github require it to be a report.